crates/interpreter/src/instructions/macros.rs

```diff
@@ -49,32 +49,34 @@ macro_rules! gas_or_fail {
     };
 }
 
-macro_rules! memory_resize {
-    ($interp:expr, $offset:expr, $len:expr) => {
+macro_rules! shared_memory_resize {
+    ($interp:expr, $offset:expr, $len:expr) => {{
+        let len: usize = $len;
+        let offset: usize = $offset;
```

Comment:
Here I'm creating another inner scope to allow a `borrow_mut`. I'd be better to use the `mem` variable also elsewhere in the opcode which calls this macro but I get some compilation problems.. I'd like to get the same result of the `pop!` macro but I can't... My Rust chops are not very good yet. How can I achieve this?
I wonder if this has a particular performance penalty